### PR TITLE
[ci] Use sign-artifacts template for macOS signing

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/yaml-templates/pull/317

We've been hitting a lot of issues recently in the macOS developer
signing steps that run during the build. Mono MSBuild has been crashing,
and signing service requests have been timing out.

The macOS developer signing steps have been moved to the post build
signing job that runs on Windows agents to try to mitigate these issues.

### CODE REVIEW

**Comment by pjcollins** [comment]
Finally managed to get through all the issues and various outages, and the latest test build looks good: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9889091&view=results

Checking a few files randomly in the Darwin pack the signatures seems right:

> tools % codesign -dv --verbose=4  Darwin/aapt2
Executable=/Users/peter/Downloads/Microsoft.Android.Sdk.Darwin.35.0.0-ci.dev-pjc-macsign-template.21/tools/Darwin/aapt2
Identifier=aapt2-555549444c4c446e55553144a1a4e01cffdb06ce
Format=Mach-O universal (x86_64 arm64)
...
Authority=Developer ID Application: Microsoft Corporation 
Authority=Developer ID Certification Authority
Authority=Apple Root CA

> tools % codesign -dv --verbose=4  Darwin/binutils/lib/liblldCOFF.dylib 
Executable=/Users/peter/Downloads/Microsoft.Android.Sdk.Darwin.35.0.0-ci.dev-pjc-macsign-template.21/tools/Darwin/binutils/lib/liblldCOFF.dylib
Identifier=liblldCOFF
Format=Mach-O universal (x86_64 arm64)
...
Authority=Developer ID Application: Microsoft Corporation
Authority=Developer ID Certification Authority
Authority=Apple Root CA

File: build-tools/create-packs/SignList.xml [hunk 1]
```diff
@@ -26,5 +26,12 @@
     <FirstParty Include="*.exe" />
     <!-- First and third party .jar files use the same Authenticode -->
     <FirstParty Include="*.jar" />
+    <MacDeveloperSign Include="*.dylib" />
+    <MacDeveloperSignHarden Include="Darwin/aapt2" />
+    <MacDeveloperSignHarden Include="Darwin/binutils/bin/*as" />
+    <MacDeveloperSignHarden Include="Darwin/binutils/bin/*ld" />
+    <MacDeveloperSignHarden Include="Darwin/binutils/bin/llc" />
+    <MacDeveloperSignHarden Include="Darwin/binutils/bin/llvm-mc" />
+    <MacDeveloperSignHarden Include="Darwin/binutils/bin/*strip" />
```
Comment (jonathanpeppers) [inline build-tools/create-packs/SignList.xml:35]: Previously, we used an item group like `@(_MSBuildFilesUnixSignAndHarden)` or `@(_BinUtilsFilesUnixSign)`, is there a chance we might miss a file?

**Review by jonathanpeppers — COMMENTED** [review]
Will we have any issues taking this back to .NET 8? Wondering how many total commits would be required.

**Comment by pjcollins** [comment]
> Will we have any issues taking this back to .NET 8? Wondering how many total commits would be required.

I don't think we will need to backport this, as the recent changes to use dotnet build and disable CodeQL on non-main branches should get us building again there.

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**